src/test/java/teammates/test/cases/testdriver/TestngTest.java

```diff
@@ -0,0 +1,80 @@
+package teammates.test.cases.testdriver;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.testng.annotations.Test;
+
+import teammates.common.util.FileHelper;
+import teammates.test.cases.BaseTestCase;
+
+public class TestngTest extends BaseTestCase {
+
+    HashMap<String, String> tests = new HashMap<String, String>(); // <class name, package name>
+    String testngXmlAsString;
+
+    @Test
+    public void checkTestsInTestng() throws FileNotFoundException {
+        assertTrue(FileHelper.isFileExists("./src/test/testng.xml"));
+        testngXmlAsString = FileHelper.readFile("./src/test/testng.xml");
+        
+        addFilesToTestsRecursively();
+        excludeFilesNotInTestng();
```

Comment:
Similar case here. Pass whatever it needs as parameters and get the result as a return value. That way the method becomes self-explanatory, less coupled with the rest of the code, and more testable.